File: packages/android_intent_plus/lib/src/put_classes/put_short.dart
import 'package:android_intent_plus/src/put_classes/base/put_base.dart';

class PutShort extends PutBase<int> {
  PutShort({required String key, required int value})
      : super(key: key, value: value) {
    if (value < -32768 || value > 32767) {
      throw RangeError.value(
        value,
        "value must be between -32768 and 32768, inclusive.",
      );
    }
  }

  @override
  String get javaClass => 'PutShort';

  factory PutShort.fromJson({required String key, required dynamic value}) {
    return PutShort(key: key, value: value as int);
  }

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'key': key,
      'javaClass': javaClass,
      'value': value,
    };
  }
}

File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutShort.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;
import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;
import org.json.JSONException;
import org.json.JSONObject;

public class PutShort extends PutBase<Short> {

  public static final String JAVA_CLASS = "PutShort";

  public PutShort(String key, Short value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutShort putShort) {
    bundle.putShort(putShort.key, putShort.value);
  }

  public static PutShort fromJson(JSONObject jsonObject) throws JSONException {
    return new PutShort(
        jsonObject.getString(Constants.KEY), (short) jsonObject.getInt(Constants.VALUE));
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", value);
    return jsonObject;
  }
}
